src/functional/fencil_processors/codegens/gtfn/itir_to_gtfn_ir.py

```diff
@@ -59,24 +61,48 @@ class GTFN_lowering(eve.NodeTranslator, eve.VisitorWithSymbolTableTrait):
     }
     _unary_op_map = {"not_": "!"}
 
+    def __init__(self):
+        self.offset_definitions: dict[str, common.Dimension | common.Connectivity] = {}
+
     def visit_Sym(self, node: itir.Sym, **kwargs: Any) -> Sym:
         return Sym(id=node.id)
 
     def visit_SymRef(self, node: itir.SymRef, **kwargs: Any) -> SymRef:
         return SymRef(id=node.id)
 
     def visit_Lambda(self, node: itir.Lambda, **kwargs: Any) -> Lambda:
-        return Lambda(params=self.visit(node.params), expr=self.visit(node.expr))
+        return Lambda(params=self.visit(node.params), expr=self.visit(node.expr, **kwargs))
 
     def visit_Literal(self, node: itir.Literal, **kwargs: Any) -> Literal:
         return Literal(value=node.value, type=node.type)
 
+    _vertical_dimension = "unstructured::dim::vertical"
+    _horizontal_dimension = "unstructured::dim::horizontal"
+
     def visit_OffsetLiteral(self, node: itir.OffsetLiteral, **kwargs: Any) -> OffsetLiteral:
```

Comment:
If you have two functions:
- one for the acquisition of `offset_definitions`
- another one for node translation

then maybe the branching logic can be simpler in both. I haven't tried to rewrite it, just a suggestion.
This would also help in moving offset def collection out of this class.

Reply:
Did this, but had to change the logic: instead of renaming offset -> dimension, I keep the offset name and introduce an alias instead.